data/modules/SecondHand/SecondHand.lua

```diff
@@ -14,116 +14,166 @@ local PlayerState = require 'PlayerState'
 local utils = require 'utils'
 
 local l = Lang.GetResource("module-secondhand")
-local l2 = Lang.GetResource("ui-core")
 
 -- average number of adverts per BBS and billion population
 local N_equil = 0.1 -- [ads/(BBS*unit_population)]
 
 -- inverse half life of an advert in (approximate) hours
-local inv_tau = 1.0 / (4*24)
+local inv_tau = 1.0 / (4 * 24)
 
+-- Maximum indeces for randomised messages; see data/lang/module-secondhand/en.json
 -- Note: including the 0
 local max_flavour_index = 5
 local max_surprise_index = 2
+local max_money_index = 3
 
 local flavours = {}
-for i = 0,max_flavour_index do
-    table.insert(flavours, {
-        adtitle = l["FLAVOUR_" .. i .. "_TITLE"],
-        adtext = l["FLAVOUR_" .. i .. "_TEXT"],
-        adbody  = l["FLAVOUR_" .. i .. "_BODY"],
-    })
+for i = 0, max_flavour_index do
+	table.insert(flavours, {
+		adtitle = l["FLAVOUR_" .. i .. "_TITLE"],
+		adtext  = l["FLAVOUR_" .. i .. "_TEXT"],
+		adbody  = l["FLAVOUR_" .. i .. "_BODY"],
+	})
 end
 
 local ads = {}
 
 -- find a value in an array and return key
-local onDelete = function (ref)
-    ads[ref] = nil
+local onDelete = function(ref)
+	ads[ref] = nil
 end
 
--- check it fits on the ship, both the slot for that equipment and the
--- weight.
+-- Check if a piece of equipment can fit into the ship. Both the available space
+-- and, for equipemnt which requires a slot, a free and compatible slot are
+-- checked.
+--
+-- Returns a triple of:
+--
+--   ok      - whether the equipment can be installed
+--
+--   slot?   - if ok and the equipment requires a slot, the free slot into which
+--             it can be installed
+--
+--   message - a message containing the reason why the equipment can't be installed
+--
 ---@param e EquipType
-local canFit = function (e)
-
+---@return boolean,Hullconfig.Slot?,string
```

Comment:
Nitpick: prefer documenting multiple returns as:
```suggestion
---@return boolean ok
---@return Hullconfig.Slot? slot
---@return string error
```

Reply:
Fixed (and squashed)